src/fields/jsgrid.field.control.js

```diff
@@ -91,16 +93,16 @@
         },
 
         filterTemplate: function() {
-            var $result = this._createSearchButton();
+            var $result = this.searchButton ? this._createSearchButton() : $("");
             return this.clearFilterButton ? $result.add(this._createClearFilterButton()) : $result;
         },
 
         insertTemplate: function() {
-            return this._createInsertButton();
+            return this.insertButton ? this._createInsertButton() : $("");
```

Comment:
For insert button (the same as for update button): how user can insert an item without this button?
If we doesn't need inserting, he can turn it off with `inserting` option.